### DISCUSSION TITLE
Is it possible to demote a Distributed Manager Search node to just a Manager Node? So that i can add multiple receiver nodes, but keep all my work?

### DISCUSSION THREAD

**Discussion by Phillip-Samson** [Security-Onion-Solutions/securityonion#11992] [2.4] [answered]
### Version

2.4.30

### Installation Method

Network installation on Red Hat derivative like Oracle, Rocky, Alma, etc.

### Description

configuration

### Installation Type

Distributed

### Location

cloud

### Hardware Specs

Exceeds minimum requirements

### CPU

16

### RAM

64

### Storage for /

250

### Storage for /nsm

30tb

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

more than 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

With the distributed Manager Search node the receiver node failed highstate with multiple docker errors.
Example
----------
          ID: so-redis
    Function: docker_container.running
      Result: False
     Comment: Failed to pull sosewrver:5000/security-onion-solutions/so-redis:2.4.30: Error 404: manifest for soserver:5000/security-onion-solutions/so-redis:2.4.30 not found: manifest unknown: manifest unknown
     Started: 20:24:07.493424
    Duration: 74.647 ms
     Changes:
     
     
If it is not possible to use a receiver node with a manager search node, is it possible to demote the MS to just a manager so that I can bolt on multiple receivers?
     
     
     

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by TOoSmOotH** [comment]
Receiver nodes do not run elasticsearch. What are you really trying to do? It makes no sense to add redundancy to the pipeline if you don't have multiple search nodes. Receiver nodes are for pipeline redundancy.

**Reply by Phillip-Samson** [reply]
I  created the distributed manage search node to scale from the single instance to a more adequately distributed architecture as needed over time, starting small and growing. As things develop, we are looking to offset some of the compute to a separate receiver, expecting to see better performance on the manager search node while running searches ( I reserve the right to be wrong with this thinking :)
We have expanded our need to include an isolated sister company. We need/want to manage both companies from a single UI vi, a VPN connecting to their Azure instances of receiver and search nodes.
To test my theory and offset some of company A's computing, I set up a receiver node expecting some of the traffic sent to the manager search to migrate to the new receiver. Still, I found that none of the SO services were running on the RECEIVER, and in my attempt to find out if what I was trying to do was even possible, I was informed that it might not be possible. So, if it is impossible to configure this type of setup Using a manager search node, I need to break it all apart. I need to know if it will likely demote my manager search to a simple manager, stand up a search node and a receiver node, and then continue to expand.

**Reply by dougburks** [reply]
> we are looking to offset some of the compute to a separate receiver, expecting to see better performance on the manager search node while running searches ( I reserve the right to be wrong with this thinking :)

It seems like you may be confusing search nodes and receiver nodes.

From https://docs.securityonion.net/en/2.4/architecture.html#search-node:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/1659467/baa1fad6-2cfa-4f0c-9f09-f9233de4fff0)

From https://docs.securityonion.net/en/2.4/architecture.html#receiver-node:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/1659467/95487f8b-537e-4b31-899a-df39a68da06b)

**Reply by Phillip-Samson** [reply]
OK, 
This goes back to my original question is it possible to use a Receiver and Search node with a Distributed Manger+Search Node
or do i need to breake the Manager Search node apart now to be able to expand?
the below diagram is what I am attempting to build. Will this work?
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36277967/0a63eefa-d035-414a-a500-88aae5f27b04)


If the answer to the above question ius yes can someon help me trouble shoot the fact that the current Reciever Node i added to my cluster wont start the containers
[https://github.com/Security-Onion-Solutions/securityonion/discussions/11975]

Current Receiver Node Status
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36277967/27eda57b-2400-40b7-ac7f-7927ba183824)
https://github.com/Security-Onion-Solutions/securityonion/discussions/11975#discussioncomment-7821119


I am sorry , i have turned this all into a mess it seems

**Reply by kvkamin** [reply]
Make sure docker is started. 
systemctl <status/start> docker

If docker is running, the containers should start.

**Reply by Phillip-Samson** [reply]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36277967/c8315de4-8ef2-4fd2-acf1-f5a6aa9ccce9)

**Reply by TOoSmOotH** [reply]
When you run a highstate from one of the receiver nodes does it complete?

**Comment by Phillip-Samson** [comment]
Fails with four errors shown below..
----------
          ID: so-sensoroni
    Function: docker_container.running
      Result: False
     Comment: Failed to pull orseconip01:5000/security-onion-solutions/so-soc:2.4.30: Error 404: manifest for orseconip01:5000/security-onion-solutions/so-soc:2.4.30 not found: manifest unknown: manifest unknown
     Started: 15:16:05.729938
    Duration: 73.143 ms
     Changes:

----------
          ID: so-telegraf
    Function: docker_container.running
      Result: False
     Comment: Failed to pull orseconip01:5000/security-onion-solutions/so-telegraf:2.4.30: Error 404: manifest for orseconip01:5000/security-onion-solutions/so-telegraf:2.4.30 not found: manifest unknown: manifest unknown
     Started: 15:16:06.304147
    Duration: 64.593 ms
     Changes: 

----------
          ID: so-logstash
    Function: docker_container.running
      Result: False
     Comment: Failed to pull orseconip01:5000/security-onion-solutions/so-logstash:2.4.30: Error 404: manifest for orseconip01:5000/security-onion-solutions/so-logstash:2.4.30 not found: manifest unknown: manifest unknown
     Started: 15:16:08.258483
    Duration: 70.008 ms
     Changes: 
----------
          ID: so-redis
    Function: docker_container.running
      Result: False
     Comment: Failed to pull orseconip01:5000/security-onion-solutions/so-redis:2.4.30: Error 404: manifest for orseconip01:5000/security-onion-solutions/so-redis:2.4.30 not found: manifest unknown: manifest unknown
     Started: 15:16:09.514269
    Duration: 68.849 ms
     Changes:   
----------


Summary for local
--------------
Succeeded: 191 (changed=9)
Failed:      4
--------------
Total states run:     195
Total run time:    19.995 s

**Reply by TOoSmOotH** [reply]
try so-docker-refresh on the manager. If that doesn't fix it make sure there is connectivity from the receiver to the manager on port 5000: nc -vz manager 5000

**Reply by Phillip-Samson** [reply, marked as answer]
LOL after all that. Docker refresh fixed it. Thank you for your help
